htdocs/core/modules/modStock.class.php

```diff
@@ -295,10 +336,12 @@ public function __construct($db)
 		}
 		$this->export_aggregate_array[$r] = array('sm.value'=>'SUM'); // TODO Not used yet
 		$this->export_dependencies_array[$r] = array('movement'=>array('sm.rowid')); // We must keep this until the aggregate_array is used. To add unique key if we ask a field of a child to avoid the DISTINCT to discard them.
+
 		$this->export_sql_start[$r] = 'SELECT DISTINCT ';
-		$this->export_sql_end[$r]  = ' FROM '.MAIN_DB_PREFIX.'product as p, '.MAIN_DB_PREFIX.'stock_mouvement as sm, '.MAIN_DB_PREFIX.'entrepot as e';
-		$this->export_sql_end[$r] .= ' WHERE p.rowid = sm.fk_product AND sm.fk_entrepot = e.rowid';
-		$this->export_sql_end[$r] .= ' AND e.entity IN ('.getEntity('stock').')';
+		$this->export_sql_end[$r]  = ' FROM '.MAIN_DB_PREFIX.'stock_mouvement as sm';	// Primary table is stock movements
+		$this->export_sql_end[$r] .= ' LEFT JOIN '.MAIN_DB_PREFIX.'product as p ON p.rowid = sm.fk_product';
```

Comment:
Same. Main table is stock_mouvement so product and entrepot are always linked. INNER join is better to let the database choose the best execution plan according on filter used.

Reply:
Hi,

I've used LEFT join for entrepot because there is no foreign key constraint.
So what happens if the user deletes a warehouse ?
Does the code guarantee database integrity, so that we can safely use INNER join  ?

> Old sql code looks better.

Sure, it can look better, but this old syntax is "deprecated" since SQL-92 standard because it's not explicit, so ambiguous.
https://stackoverflow.com/questions/894490/sql-left-join-vs-multiple-tables-on-from-line

There's no recommendation for JOINs on the wiki: 
https://wiki.dolibarr.org/index.php?title=Langages_et_normes#R.C3.A8gles_de_codage_SQL

Should I revert to old syntax ?